independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanInfo.java

```diff
@@ -419,7 +420,14 @@ void init(List<Throwable> errors, Consumer<BytecodeTransformer> bytecodeTransfor
         if (disposer != null) {
             disposer.init(errors);
         }
-        interceptedMethods.putAll(initInterceptedMethods(errors, bytecodeTransformerConsumer, removeFinalForProxyableMethods));
+
+        synchronized (lock) {
```

Comment:
Why the lock, the volatile field and all the magic here?

Reply:
`interceptedMethods` being originally a `ConcurrentHashMap` suggested the possibility that `init()` might be called multiple times from concurrent threads. So I over protected it to see if it is going to fail somewhere. I am open to simplify it.

Reply:
BTW `interceptedMethods` needs to be a LinkedHashMap for the sake of reproducibility.

Reply:
`init()` is only called once. I'm not sure about the `ConcurrentHashMap` though - I'll verify this when I have some spare time.